analyses/snv-callers/compare_snv_callers.Rmd

```diff
@@ -0,0 +1,484 @@
+}
+# Make caller specific plots folder
+if (!dir.exists(consensus_plots_dir)) {
+  dir.create(consensus_plots_dir)
+}
+```
+
+## Import data 
+
+Look at the existing folders within the results folder. 
+
+```{r}
+# Exclude the non-caller directories
+caller_dirs <- grep("vaf_cutoff|consensus", dir(base_results_dir, full.names = TRUE), invert = TRUE, value = TRUE)
+
+# Print this out to check
+caller_dirs
+```
+
+Get the lists of each type of file. 
+
+```{r}
+# Get a list of vaf files
+vaf_files <- sapply(caller_dirs, 
+                    list.files, pattern = "_vaf", recursive = TRUE, full.names = TRUE)
+
+# Print this out to check
+vaf_files
+```
+
+```{r}
+# Get a list of tmb files
+tmb_files <- sapply(caller_dirs, 
+                    list.files, pattern = "_tmb", recursive = TRUE, full.names = TRUE)
+
+# Print this out to check
+tmb_files
+```
+
+Obtain the caller names from the file paths. 
+
+```{r}
+# Get the caller names
+caller_names <- stringr::word(vaf_files, sep = "/", 2)
+```
+
+Read in VAF files. 
+
+```{r include=FALSE}
+# Read in the other files to match the first
+vaf_list <- lapply(vaf_files, function(vaf_file) {
+  # Read in the file
+  df <- readr::read_rds(vaf_file) 
+  
+  # Make it so it is more easily combined with the other files
+  df %>%
+    # Attempt to make numeric columns where that doesn' kick back an "NA"
+    dplyr::mutate_at(dplyr::vars(which(!is.na(as.numeric(t(df[1,]))))), as.numeric) %>% 
+    # Aliquot id sometimes contains letters and sometimes numbers across the callers
+    dplyr::mutate(aliquot_id = as.character(aliquot_id), 
+                  variant_qual = as.character(variant_qual)) %>% 
+    # Turn these columns into characters because otherwise they cause trouble.
+    dplyr::mutate_at(dplyr::vars(dplyr::contains("AF", ignore.case = FALSE)), as.character) %>% 
+    # Get rid of the few if any duplicate entries. 
+    dplyr::distinct(mutation_id, .keep_all = TRUE)
+})
+
+# Carry over the callers' names
+names(vaf_list) <- caller_names
+```
+
+Make a big VAF data.frame
+
+```{r include=FALSE}
+vaf_df <- dplyr::bind_rows(vaf_list, .id = "caller") %>%
+  dplyr::mutate(caller = factor(caller))
+```
+
+Read in TMB files. 
+
+```{r include=FALSE}
+tmb_list <- lapply(tmb_files, readr::read_rds)
+
+# Carry over the callers' names
+names(tmb_list) <- caller_names
+```
+
+Make a big tmb data.frame
+
+```{r include=FALSE}
+tmb_df <- dplyr::bind_rows(tmb_list, .id = "caller") %>%
+  dplyr::mutate(caller = factor(caller))
+```
+
+## Upset graph
+
+```{r}
+mutation_id_list <- lapply(vaf_list, function(caller) caller$mutation_id)
+```
+
+Make the upset plot. 
+
+```{r}
+# Make the plot
+upset <- UpSetR::upset(UpSetR::fromList(mutation_id_list),
+  order.by = "freq",
+  text.scale = 2
+)
+
+# Save this plot
+png(file.path(consensus_plots_dir, "upset_plot.png"), width = 900, height = 900)
+upset
+dev.off()
+
+# Print this out
+upset
+```
+
+## VAF for combination sets of callers
```

Comment:
I feel like this calculation & the actual combination of these callers should be in a separate script. While we hope not to have to run this repeatedly, it might be nice to run the consensus generation separately from the full notebook.